Chrysalis-Papilio/Soen-490-Papilio-Android
# User Activity creation

# General info

**Issue number**: PAPILIO-23

**Task description**: 

 - User should be able to create new activities using their mobile devices

# Testing
To test this feature you have to:

0. Initially, the code will not compile because a mapbox token is required: you need to place this line in **local.properties** file: MAPBOX_SECRET_TOKEN=mapbox_token (contact me to obtain the token)
1. Navigate to the CreateActivity page & create a valid activity & verify that images are properly uploaded in Firebase storage and that the correct entry appears in the database
2. Try to create an activity and omit a field, it should show a snackbar
3. Verify that mapbox gives address suggestions when the user starts to type in the activity location field

# Notes
- @mth-1012 Still needs to fix a little issue (where he awaits an integer but form data can only be of string type) on his route. Activity creation on Android will not work until this is fixed
